src/System/Process/Typed.hs

```diff
@@ -933,13 +937,13 @@ readProcessStderr_ pc =
   where
     pc' = setStderr byteStringOutput pc
 
-withProcessInterleave
-  :: ProcessConfig stdin stdoutIgnored stderrIgnored
-  -> (Process stdin (STM L.ByteString) () -> IO a)
-  -> IO a
+withProcessInterleave :: (MonadUnliftIO m)
+  => ProcessConfig stdin stdoutIgnored stderrIgnored
+  -> (Process stdin (STM L.ByteString) () -> m a)
+  -> m a
 withProcessInterleave pc inner =
     -- Create a pipe to be shared for both stdout and stderr
-    bracket P.createPipe (\(r, w) -> hClose r >> hClose w) $ \(readEnd, writeEnd) -> do
+    bracket (liftIO P.createPipe) (\(r, w) -> hCloseLiftIO r >> hCloseLiftIO w) $ \(readEnd, writeEnd) -> do
```

Comment:
The `liftIO`s should be unnecessary here

Reply:
Without the lift, `before` would be `IO a`, then type for `thing` cannot match.